meerk40t/core/elements.py

```diff
@@ -5540,6 +5541,79 @@ def contains(box, x, y=None):
         self._emphasized_bounds = None
         self.set_emphasis(None)
 
+    def classify_legacy(self, elements, operations=None, add_op_function=None):
+        """
+        Classify does the placement of elements within operations.
+        "Image" is the default for images.
+        Typically,
+        If element strokes are red they get classed as cut operations
+        If they are otherwise they get classed as engrave.
+        However, this differs based on the ops in question.
+        :param elements: list of elements to classify.
+        :param operations: operations list to classify into.
+        :param add_op_function: function to add a new operation, because of a lack of classification options.
+        :return:
+        """
+        if elements is None:
+            return
+        reverse = self.context.classify_reverse
+        if reverse:
+            elements = reversed(elements)
+        if operations is None:
+            operations = list(self.ops())
+        if add_op_function is None:
+            add_op_function = self.add_op
+        for element in elements:
+            was_classified = False
+            if hasattr(element, "operation"):
+                add_op_function(element)
+                continue
+            if element is None:
+                continue
+            for op in operations:
+                if op.operation == "Raster":
+                    if element.stroke is not None and op.color == abs(element.stroke):
+                        op.add(element, type="opnode")
+                        was_classified = True
+                    elif isinstance(element, SVGImage):
+                        op.add(element, type="opnode")
+                        was_classified = True
+                    elif isinstance(element, SVGText):
+                        op.add(element)
+                        was_classified = True
+                    elif element.fill is not None and element.fill.argb is not None:
+                        op.add(element, type="opnode")
+                        was_classified = True
+                elif (
+                    op.operation in ("Engrave", "Cut")
+                    and element.stroke is not None
+                    and op.color == abs(element.stroke)
+                ):
+                    op.add(element, type="opnode")
+                    was_classified = True
+                elif op.operation == "Image" and isinstance(element, SVGImage):
+                    op.add(element, type="opnode")
+                    was_classified = True
+                    break  # May only classify in one image operation.
+                elif (
```

Comment:
This should use the isDots to determine whether they are dots as there is more than one way to get a dot.

Reply:
Can't legacy code.

Reply:
Dots is NOT in the legacy code. Anywhere. It is an MK7 only thing. And maintaining compatibility with a barely functional 0.7 RC1 used only by a couple of hard-core testers is completely different to maintaining compatibility with 0.6.2x which was widely used for a long time by the MK community.